lang/cs/Org.Apache.REEF.Driver/Bridge/Events/EvaluatorRequestor.cs

```diff
@@ -63,13 +67,24 @@ internal EvaluatorRequestor(IEvaluatorRequestorClr2Java clr2Java)
 
         public void Submit(IEvaluatorRequest request)
         {
-            LOGGER.Log(Level.Info, "Submitting request for {0} evaluators and {1} MB memory and  {2} core to rack {3}.", request.Number, request.MemoryMegaBytes, request.VirtualCore, request.Rack);
+            LOGGER.Log(Level.Info, "Submitting request for {0} evaluators and {1} MB memory and  {2} core to rack {3} and runtime {4}.", request.Number, request.MemoryMegaBytes, request.VirtualCore, request.Rack, request.RuntimeName);
 
+            byte[] data = Clr2Java.GetDefinedRuntimes();
+            DefinedRuntimes runtimes = DefinedRuntimesSerializer.FromBytes(data);
```

Comment:
No Julia,

We want to move the validation to the C# side, as it will make C# troubleshooting easier. The other possibility is to make exception throwing from java to c# better, but this is much biger work item as we need to do that across all interfaces.